sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryContext.java

```diff
@@ -7,43 +7,52 @@
 
 import java.time.Duration;
 import java.time.Instant;
+import java.util.ArrayList;
 import java.util.List;
 
 public class RetryContext {
-
-    @JsonIgnore
-    public List<int[]> directRetrySpecificStatusAndSubStatusCodes;
-    @JsonIgnore
-    public List<int[]> genericRetrySpecificStatusAndSubStatusCodes;
-
     @JsonIgnore
-    public Instant retryStartTime;
+    public volatile Instant retryStartTime;
```

Comment:
I am not sure of what exactly is the intent behind making this volatile, but just a point to consider: volatile will not guarantee thread safety in all the scenarios

Reply:
Volatile is what we using across retry policy for multiple thread to change single value. Want to avoid dirty read

Reply:
If the case is just to avoid dirty read, then I think we are good. Thanks